pkg/cmd/cluster/cluster_list.go

```diff
@@ -20,62 +20,82 @@ func newCmdClusterList(ctx api.Context) *cobra.Command {
 		Use:  "list",
 		Args: cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			clusters := ctx.Clusters()
-
 			currentCluster, err := ctx.Cluster()
 			var currentClusterID string
 			if err == nil {
 				currentClusterID = currentCluster.ID()
 			}
 
-			// infos is composed of clusters information: attachment, name, id, status, version, and url.
-			infos := []clusterInfo{}
+			var clusters []*cli.Cluster
+			if !onlyAttached {
+				clusters = ctx.Clusters()
+			} else if currentCluster != nil {
+				clusters = append(clusters, currentCluster)
+			}
+
+			// ClusterInfo contains information about a cluster, it represents an item in the list.
+			type ClusterInfo struct {
+				Attached bool   `json:"attached"`
+				ID       string `json:"id"`
+				Name     string `json:"name"`
+				Status   string `json:"status"`
+				URL      string `json:"url"`
+				Version  string `json:"version"`
+			}
+
+			var items []ClusterInfo
 			var mu sync.Mutex
 
 			var wg sync.WaitGroup
 			for _, cluster := range clusters {
-				// Add should execute before the statement creating the goroutine.
 				wg.Add(1)
 				go func(cluster *cli.Cluster) {
 					defer wg.Done()
-
-					var info clusterInfo
-					info.Name = cluster.Name()
-					info.ID = cluster.ID()
-					if info.ID == currentClusterID {
-						info.Attached = true
-					} else if onlyAttached {
-						return
-					}
-					info.URL = cluster.URL()
-					info.Status = "AVAILABLE"
-					info.Version, err = getVersion(ctx, cluster)
-					if err != nil {
-						info.Status = "UNAVAILABLE"
+					item := ClusterInfo{
+						ID:       cluster.ID(),
+						Name:     cluster.Name(),
+						URL:      cluster.URL(),
+						Status:   "UNAVAILABLE",
+						Version:  "UNKNOWN",
+						Attached: (cluster.ID() == currentClusterID),
 					}
 
+					httpClient := ctx.HTTPClient(cluster, httpclient.Timeout(5*time.Second))
+					version, err := cli.NewDCOSClient(httpClient).Version()
+					if err == nil {
+						item.Status = "AVAILABLE"
+						item.Version = version.Version
+					}
 					mu.Lock()
-					infos = append(infos, info)
-					mu.Unlock()
+					defer mu.Unlock()
+
+					// Order clusters by name, this guarantees a stable list in table output.
+					itemsLen := len(items)
```

Comment:
I find this loop slightly more comprehensible:
```
// Order clusters by name, this guarantees a stable list in table output.
for i := range items {
	if items[i].Name > item.Name {
		items = append(items[:i], append([]ClusterInfo{item}, items[i:]...)...)
		return
	}
}
```
I ran the tests with that loop and it passed.

Reply:
definitely :+1:  just updated it